XenAdmin/Wizards/PatchingWizard/PlanActions/PatchPrechecksOnMultipleHostsAction.cs

```
@@ -0,0 +1,116 @@
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING 
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF 
+ * SUCH DAMAGE.
+ */
+
+using System;
+using System.Collections.Generic;
+using XenAdmin.Actions;
+using XenAdmin.Core;
+using XenAPI;
+using System.Linq;
+using System.IO;
+using XenAdmin.Network;
+using XenAdmin.Diagnostics.Checks;
+
+namespace XenAdmin.Wizards.PatchingWizard.PlanActions
+{
+    class PatchPrechecksOnMultipleHostsPlanAction : PlanActionWithSession
+    {
+        private readonly XenServerPatch patch;
+        private readonly List<PoolPatchMapping> mappings;
+        private List<Host> hosts = null;
+
+        public PatchPrechecksOnMultipleHostsPlanAction(IXenConnection connection, XenServerPatch patch, List<Host> hosts, List<PoolPatchMapping> mappings)
+            : base(connection, string.Format("Checking free space for uploading and installing {0}...", patch.Name))
+        {
+            this.patch = patch;
+            this.hosts = hosts;
+            this.mappings = mappings;
+        }
+
+        protected override void RunWithSession(ref Session session)
+        {
+            var mapping = mappings.Find(m => m.XenServerPatch == patch);
+            if (mapping != null && mapping.Pool_patch != null)
+            {
+
+                foreach (var host in hosts)
+                {
+                    try
+                    {
+                        var check = new PatchPrecheckCheck(host, mapping.Pool_patch);
+                        var problem = check.RunCheck();
+                        
+                        if (problem != null)
+                        {
+                            //there is a problem...

```

We need to throw an exception here if there is a problem. Or report the problem somehow.